paint/PAINT_ColourPicker.cpp

```diff
@@ -117,48 +189,25 @@ void ColourPicker::draw()
 	SDL_RenderFillRect(renderer_, &boxRect);
 	
 	//update the colour displays to match colour setup in drawWindow
-	primaryColourDisplay_->updateColour(drawWindowPtr_->getPrimaryColour());
-	secondaryColourDisplay_->updateColour(drawWindowPtr_->getSecondaryColour());
+	primaryColourDisplay_->setColour(drawWindowPtr_->getPrimaryColour());
+	secondaryColourDisplay_->setColour(drawWindowPtr_->getSecondaryColour());
 	
 	 for (auto const& child : getChildren())
 	{
 		child->draw();
 	}
 }
 
-void ColourPicker::mouseButtonDown(win::MouseButton const button)
+bool ColourPicker::mouseButtonDown(win::MouseButton const button)
 {
 	std::cout << "Clicking down in ColourPicker area\n";
```

Comment:
Don't leave debug printing in for all builds. Use `#if defined(_DEBUG)` to ensure it doesn't pollute release code.
**MUST**

Reply:
done, and removed mouseButtonDown/Up for ColourPicker class, since nothing should happen if clicking on the ColourPicker space that is not one of its child UIelements.